### DISCUSSION TITLE
Consume folder

### DISCUSSION THREAD

**Discussion by nicksolyom** [ciur/papermerge#464] [Q&A] [answered]
Hello! I have the papermerge docker container installed on my synology and I'm having trouble getting the consume folder to work. 

I have created a folder called volume1/docker/papermerge/consume/ and edited the config file like so:
<img width="424" alt="Screen Shot 2022-06-27 at 9 10 32 PM" src="https://user-images.githubusercontent.com/21064054/176090638-f3daf505-07d8-472f-a559-40ce5affb84f.png">

I'm not getting any results, thank you in advance for any help

**Comment by xe2io** [comment]
Hi, I'm not that familiar with how Synology has implemented its Docker/volume management, but it sounds like you may need to define a [volume mount](https://docs.docker.com/storage/volumes/) from the host (nas) to the papermerge container.
Presumably, you may already have something like:
`volume1/docker/papermerge/media` mapped to `/app/papermerge/static`
and would need to similarly define one for the importer directory:
`volume1/docker/papermerge/consume` mapped to `/consume`

Hope this helps in case you haven't resolved your issue yet.

**Reply by nicksolyom** [reply]
Thank you for taking the time to help! Here's my mounting set up in docker in synology, I thought I had it right but not getting any documents in my inbox.
![Screenshot 2022-08-02 215106](https://user-images.githubusercontent.com/21064054/182506962-e3299073-6d96-45e2-b6a4-4c835e26b812.png)
.

**Comment by xe2io** [comment, marked as answer]
You're welcome, and apologies for the late reply.  I did a quick test with volume mounts and the [linuxserver/papermerge](https://hub.docker.com/r/linuxserver/papermerge) container and it looks like it was ingesting[^1].  The ingest should be going to the [superuser's inbox](https://papermerge.readthedocs.io/en/v2.0.1/consumption.html#the-importer-directory) if it's working properly; can you check either the container logs (usually on stdout) or if there are OCR start/stop messages in the admin logs?
  
A couple other things to check:
- Which papermerge image are you using?  I assumed the linuxserver.io one since that's the one in the docker-compose examples, although I think there's an official one as well.
- Have you restarted the container since modifying the papermerge.conf.py file?  The container logs should complain that the `IMPORTER_DIR` variable is not set (right before imap/mail config):
```
spawned uWSGI master process (pid: 177)
spawned uWSGI worker 1 (pid: 191, cores: 1)
[uwsgi-daemons] spawning "/usr/bin/python3 ./manage.py worker" (uid: 1000 gid: 1000)
[2022-08-04 04:52:41,865: WARNING/MainProcess] Importer from local folder task not started.Reason:
1. IMPORTER_DIR not configured
2. importer dir does not exist

[2022-08-04 04:52:41,865: WARNING/MainProcess] Importer from imap account not started.Reason:
PAPERMERGE_IMPORT_MAIL_USER not defined
PAPERMERGE_IMPORT_MAIL_HOST not defined
```
- (less likely) If you're running different containers for the papermerge components, make sure `IMPORTER_DIR` is defined in the worker's papermerge.conf.py
- If you have shell access on the NAS, you could try execing into the container to look at the contents and permissions of `/consume`
  
Here's a [repo](https://github.com/xe2io/papermerge_importerdir_test) for the test environment if anyone would like to replicate.

[^1]: The logs (stdout) show the worker was processing the file and it was showing up in the superuser's inbox, but the file permissions meant the worker couldn't move the file so copies continually showed up in the inbox :P

**Reply by nicksolyom** [reply]
The answer was that the ingest was going to the superuser, not my local user! The first thing I did after creating the container was to create another user account for myself to use besides the default, which wasn't showing anything in the inbox. Thank you!

**Comment by xe2io** [comment]
You're welcome, and glad you got it sorted out!  Creating a non-superuser
account to use is a good practice _and_ the right thing to do.

It should be possible to create a folder for the automations and then share
it with a user or group so that normal users are able to see the results.
